assets/js/blocks/product-query/inspector-controls/upgrade-notice.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Notice, Button } from '@wordpress/components';
+
+const FormattedNotice = ( { notice }: { notice: string } ) => {
+	const strongText = 'Product Collection';
+	const [ before, after ] = notice.split( strongText );
+
+	return (
+		<>
+			{ before }
+			<strong>{ strongText }</strong>
+			{ after }
+		</>
+	);
+};
+
+export const UpgradeNotice = ( props: { upgradeBlock: () => void } ) => {
+	const notice = __(
+		'Upgrade all Products (Beta) blocks on this page to Product Collection for more features!',
+		'woo-gutenberg-products-block'
+	);
+
+	const buttonLabel = __(
+		'Upgrade to Product Collection',
+		'woo-gutenberg-products-block'
+	);
+
+	const handleClick = () => {
+		props.upgradeBlock();
+	};
```

Comment:
Really nitpicking here, but perhaps we can improve readibility a bit by destructuring props in the function signature?
```suggestion
export const UpgradeNotice = ( { upgradeBlock }: { upgradeBlock: () => void } ) => {
	const notice = __(
		'Upgrade all Products (Beta) blocks on this page to Product Collection for more features!',
		'woo-gutenberg-products-block'
	);

	const buttonLabel = __(
		'Upgrade to Product Collection',
		'woo-gutenberg-products-block'
	);

	const handleClick = () => {
		upgradeBlock();
	};
```

Reply:
It's a valid suggestion, but personally, I prefer the current state. Unless you have a strong opinion on this I'd keep the current version and the reason behind is it's aligned with the general convention used in the project. (I mean, I know we don't have a rule for that, but I think the majority of blocks don't destructure `props`, although it's an assumption, not fact 😅 ).

Reply:
Thanks for addressing my comment!

I don't have a strong opinion either. Feel free to leave it as it is :)